src/VideoCommand.cc

```diff
@@ -243,31 +268,16 @@ Json::Value FindVideo::construct_responses(
 {
     const Json::Value& cmd = json[_cmd_name];
 
-    Json::Value ret;
+    Json::Value ret = check_responses(responses);
```

Comment:
Shouldn't you return from here if there was an error? I suppose you have introduced this function to avoid repeating code below but the original code is to stop executing on error which I am not sure this one does. Also, if you haven't already, please do add failure test cases too.